[AMD] TunableOp take priority over DISABLE_ADDMM_HIP_LT

xw285cornell opened the pull request:
Summary: It seems super confusing that if we set DISABLE_ADDMM_HIP_LT + PYTORCH_TUNABLEOP_ENABLED, the former takes priority. This is because the former goes through the gemm_and_bias and tunable op is integrated with gemm path. Before we can integrate tunable op with gemm_and_bias, we'll probably just let tunable op takes priority

Test Plan: Run a simple linear program and verified.

Differential Revision: D56183954

jeffdaily commented:
Do you want gemm_and_bias to go through tunable, or is the fallback you've implemented here sufficient?

jeffdaily reviewed (approved):
This is a suitable workaround until TunableOp adds a gemm_and_bias implementation.

xw285cornell committed:
[AMD] TunableOp take priority over DISABLE_ADDMM_HIP_LT (#124161)

Summary:

It seems super confusing that if we set DISABLE_ADDMM_HIP_LT + PYTORCH_TUNABLEOP_ENABLED, the former takes priority. This is because the former goes through the gemm_and_bias and tunable op is integrated with gemm path. Before we can integrate tunable op with gemm_and_bias, we'll probably just let tunable op takes priority

Test Plan: Run a simple linear program and verified.

Reviewed By: nmacchioni

Differential Revision: D56183954
